src/h5web/visualizations/heatmap/utils.ts

```diff
@@ -1,8 +1,50 @@
 import { extent } from 'd3-array';
 import { rgb } from 'd3-color';
-import { scaleSequential } from 'd3-scale';
+import { scaleSequential, scaleLinear, ScaleLinear } from 'd3-scale';
 import { interpolateMagma } from 'd3-scale-chromatic';
 
+type Orientation = 'bottom' | 'left';
+
+interface OrientedAxisProps {
+  axisProps: { scale: ScaleLinear<number, number>; left: number };
+  tickLabelProps: { dx?: string | number; dy?: string | number };
+}
+
+export interface IndexAxisProps {
+  orientation: Orientation;
+  numberPixels: number;
+  width: number;
+  height: number;
+}
+
+export function getOrientedAxisProps(props: IndexAxisProps): OrientedAxisProps {
```

Comment:
Why not make two separate functions for axis props and ticklabel props ?

Reply:
Functions removed